src/cluster_legacy.c

```diff
@@ -844,31 +849,52 @@ int clusterSaveConfig(int do_fsync) {
         }
         offset += written_bytes;
     }
+    latencyEndMonitor(latency);
+    latencyAddSampleIfNeeded("cluster-config-write", latency);
 
     if (do_fsync) {
+        latencyStartMonitor(latency);
         server.cluster->todo_before_sleep &= ~CLUSTER_TODO_FSYNC_CONFIG;
         if (valkey_fsync(fd) == -1) {
             serverLog(LL_WARNING, "Could not sync tmp cluster config file: %s", strerror(errno));
             goto cleanup;
         }
+        latencyEndMonitor(latency);
+        latencyAddSampleIfNeeded("cluster-config-fsync", latency);
```

Comment:
We've only seen significant time taken in the fsync (both here and in the dir). Have you observed the other operations taking significant amounts of time?

Reply:
The other one I see most is open (10ms - 5s), which seems to try to get the dir lock something like that. And write sometimes take a few ms. I haven't seen the others in the production environment but i think i do see other take a few ms in the test environment (i dont see the unlink one). I see we have a lot latency around AOF, fsync/rename/write/fstats, so i added it all.

Reply:
Sounds good. I suppose open does also make sense. I am thinking now that since AWS uses EBS volumes, we might see the same type of latency on IO operations that other distributions see.